### DISCUSSION TITLE
Hope for new math and Statistics functions for Columns and Rows

### DISCUSSION THREAD

**Discussion by sfd99** [sqlitebrowser/sqlitebrowser#2785] [Ideas]
**sqlitebrowser** 
_is absolutely GREAT!_

As the Subject says:
   Hope for **new Math and Statistics functions** 
   for Columns and or Rows,
   integrated in  **sqlitebrowser**...

_For ex_:
yesterday, I was in **[Data Browser]** mode
and had a specific column selected.

I simply needed the 
   SUM and the AVERAGE (mean) 
of all the values in that selected column.

A Dropdown Menu in  the **[Data Browser]** mode
with some common Math and Stats Fxs
would be _VERY useful!_

Suggested Stats functions
for that Menu 
in **[Data Browser]** mode:
   **- SUM, MAX, Min, AVERAGE, MEDIAN, MODE, Q1, Q3,  _etc_.**

I use the R language,
so that would be a marriage made in heaven...

Right now,
we need to exit the DB Browser
go to a spreadsheet program,
or go to Rstudio (R program) 
to calculate and then return to DB Browser...

Hope you can add this...
_Best Greetings!._

**SFd
San Francisco
Ubuntu LINUX 20.04
R, Rstudio**

**Comment by justinclift** [comment]
Interesting.  That's not a bad idea, and is probably do-able too. :smile:

Btw, it looks like you've accidentally created a duplicate of this discussion as well, so I'll close that duplicate shortly.

**Comment by chrisjlocke** [comment]
> Right now, we need to exit the DB Browser

Just a nod that you can perform maths functions via the 'execute SQL' tab in DB4S.

![image](https://user-images.githubusercontent.com/7872218/126945433-c4fe902b-673d-4559-a8ed-04fd8289604f.png)

So to calculate the sum of field a:

    select sum(field_a)
    from table_a

![image](https://user-images.githubusercontent.com/7872218/126945526-a19ef394-cc2b-44d1-9de1-eb8969b482ae.png)

**Comment by chrisjlocke** [comment]
Also, if you select cells in the 'Browse data' tab, the status bar has maximum, sum, minimum and average values.

![image](https://user-images.githubusercontent.com/7872218/126945716-56ffe5f2-94c0-44b0-93a6-2789e9891d84.png)

**Reply by justinclift** [reply]
Oh, that's cool.  Didn't realise DB4S did that. :grin:

**Reply by MKleusberg** [reply]
Just an extra note: In case this doesn't work for you, it's likely because you're selecting too many rows. For performance reasons we only calculate these values if you select less rows than configured here:
![Screenshot_20210726_144108](https://user-images.githubusercontent.com/3153504/126990493-1434ec6f-9be0-4323-9580-ea92a2a0fd22.png)

Also make sure to not hover any toolbar button or similar after selecting the cells because this overrides the status bar with some help text.

**Reply by sfd99** [reply]
_Thanks all,_
   Chris, Martin and Justin
for the good advice and tips!.

(After **selecting** a Column,
I hadn't noticed the stats: ave, max, min,... info
@ bottom of screen!).

**Q:** 
Is there a way to select _MULTIPLE_ cols
in [Browse Data] and get stats for each col?.

**+Suggestion:**
It would still be _very useful_
to allow choosing several _typical stats fxs_
applied to user-selected col(s),
but in the  [Browse Data] tab.
(faster than typing in the [Execute SQL] tab!...).

I was thinking of a dropDown Menu w/Math Fxs
in the [Browse Data] tab,
or (daring alert!):
user types the chosen Math/Stats fxs
in the "Filter in any column field",
as comma-sep entries:
   ie:   
SUM, MAX, Min, AVERAGE, MEDIAN, MODE, Q1, Q3, etc.

Anyhow,
you guys deserve my greatest admiration 
for your work!.
Very useful, clear and simple to use.

**SFd
San Francisco
DB Browser for SQLite 3.12.2
Ubuntu LINUX 20.04
R, Rstudio**
<img width="426" alt="RSQL" src="https://user-images.githubusercontent.com/3316239/127035921-3709ebd2-0be4-46ff-a847-86142074c972.png">

Irene Steves:  **Using SQL in Rstudio**
https://irene.rbind.io/post/using-sql-in-rstudio/

above image credit to Ana Milana:
https://anamilana.com/how-to-start-practising-sql-for-data-analysis/
(using DB Browser SQLite, of course!).
